core/src/ast_builder/order_by_expr.rs

```diff
@@ -0,0 +1,82 @@
+use {
+    super::ExprNode,
+    crate::{
+        ast::{Expr, OrderByExpr},
+        parse_sql::parse_order_by_expr,
+        result::{Error, Result},
+        translate::translate_order_by_expr,
+    },
+};
+
+#[derive(Clone)]
+pub enum OrderByExprNode {
+    Expr(ExprNode, Option<bool>),
```

Comment:
How about starting with minimal spec?
It would be enough to start only with `OrderByExprNode::Text`.

```rust
Expr(ExprNode, Option<bool>),
```
This is exposed for ast-builder users and it is expected for them to use this.
Currently to use `Expr(ExprNode, Option<bool>)`, user needs to use something like this
```rust
let actual = (col("id"), Some(true)).into()
```
but I'm not sure that `Some(true)` would be simple enough for users to get it.
So.. I would like to suggest that we make this PR only including `OrderByExprNode::Text` and we can discuss `Expr` like improvement design in other issue or PR.